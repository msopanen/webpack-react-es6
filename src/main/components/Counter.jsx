import React from "react";
import { render } from "react-dom";
import PropTypes from "prop-types";
import { connect } from "react-redux";

class Counter extends React.Component {
    
    static propTypes = {
        counter: PropTypes.number,
        onIncrement: PropTypes.func,
        onDecrement: PropTypes.func,
    };

    render() {

        const { counter, onDecrement, onIncrement } = this.props;

        return (
            <div>
                <div>{counter}</div>
                <button onClick={onDecrement}>-</button>
                <button onClick={onIncrement}>+</button>
            </div>
        );
    }
};

const mapStateToProps = (state) => {
    return state;
};

const mapDispatchToProps = (dispatch) => {
    return {
        onIncrement: () => dispatch({ type: "INCREMENT" }),
        onDecrement: () => dispatch({ type: "DECREMENT" }),
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Counter);