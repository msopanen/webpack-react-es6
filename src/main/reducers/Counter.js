export default (state, action) => {
    switch (action.type) {
        case 'INCREMENT':
            return { ...state, counter: state.counter + 10 };
        case 'DECREMENT':
            return { ...state, counter: state.counter - 1 };
        default:
            return state;
    }
};