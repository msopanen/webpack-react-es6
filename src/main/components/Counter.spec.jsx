
import React from "react";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";

import Counter from "./Counter";

const middlewares = [];
const mockStore = configureMockStore(middlewares);

const getState = () => ({ counter: 0 }); // initial state of the store

it("renders initial state 0 properly", () => {
    const store = mockStore(getState);
    const tree = renderer.create(
      <Counter store={store} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
