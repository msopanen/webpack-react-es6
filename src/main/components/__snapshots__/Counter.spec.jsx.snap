// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders initial state and plus and minus 1`] = `
<IntlProvider
  locale="en"
>
  <Connect(Counter)
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(Symbol.observable): [Function],
      }
    }
  >
    <Counter
      counter={0}
      onDecrement={[Function]}
      onIncrement={[Function]}
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(Symbol.observable): [Function],
        }
      }
      storeSubscription={
        Subscription {
          "listeners": Object {
            "clear": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(Symbol.observable): [Function],
          },
          "unsubscribe": [Function],
        }
      }
    >
      <div>
        <label
          htmlFor="plus"
          id="counter"
        >
          <FormattedMessage
            defaultMessage="Count: {count}"
            id="app.home.count"
            values={
              Object {
                "count": 0,
              }
            }
          >
            <span>
              Count: 
            </span>
          </FormattedMessage>
        </label>
        <button
          id="plus"
          onClick={[Function]}
        >
          -
        </button>
        <button
          id="minus"
          onClick={[Function]}
        >
          +
        </button>
      </div>
    </Counter>
  </Connect(Counter)>
</IntlProvider>
`;

exports[`renders initial state and plus and minus 2`] = `
<IntlProvider
  locale="en"
>
  <Connect(Counter)
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(Symbol.observable): [Function],
      }
    }
  >
    <Counter
      counter={-1}
      onDecrement={[Function]}
      onIncrement={[Function]}
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(Symbol.observable): [Function],
        }
      }
      storeSubscription={
        Subscription {
          "listeners": Object {
            "clear": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(Symbol.observable): [Function],
          },
          "unsubscribe": [Function],
        }
      }
    >
      <div>
        <label
          htmlFor="plus"
          id="counter"
        >
          <FormattedMessage
            defaultMessage="Count: {count}"
            id="app.home.count"
            values={
              Object {
                "count": -1,
              }
            }
          >
            <span>
              Count: -1
            </span>
          </FormattedMessage>
        </label>
        <button
          id="plus"
          onClick={[Function]}
        >
          -
        </button>
        <button
          id="minus"
          onClick={[Function]}
        >
          +
        </button>
      </div>
    </Counter>
  </Connect(Counter)>
</IntlProvider>
`;

exports[`renders initial state and plus and minus 3`] = `
<IntlProvider
  locale="en"
>
  <Connect(Counter)
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(Symbol.observable): [Function],
      }
    }
  >
    <Counter
      counter={0}
      onDecrement={[Function]}
      onIncrement={[Function]}
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(Symbol.observable): [Function],
        }
      }
      storeSubscription={
        Subscription {
          "listeners": Object {
            "clear": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(Symbol.observable): [Function],
          },
          "unsubscribe": [Function],
        }
      }
    >
      <div>
        <label
          htmlFor="plus"
          id="counter"
        >
          <FormattedMessage
            defaultMessage="Count: {count}"
            id="app.home.count"
            values={
              Object {
                "count": 0,
              }
            }
          >
            <span>
              Count: 
            </span>
          </FormattedMessage>
        </label>
        <button
          id="plus"
          onClick={[Function]}
        >
          -
        </button>
        <button
          id="minus"
          onClick={[Function]}
        >
          +
        </button>
      </div>
    </Counter>
  </Connect(Counter)>
</IntlProvider>
`;
